File: sol/754/754.py
class Solution:
    def reachNumber(self, target: int) -> int:
        if target < 0:
            target = -target
        m = (int(sqrt(1 + 8 * target)) - 2) // 2
        q = -1
        while not (q >= 0 and (q & 3) == 0):
            m += 1
            q = m * (m + 1) - 2 * target
        return m

File: sol/754/754.cpp
struct Solution {
    int reachNumber(int target) {
        if (target < 0){
            target = -target;
        }
        long q = -1, m = (static_cast<long>(sqrt(1 + 8.0 * target)) - 2) / 2;
        while (q < 0 || (q & 3) != 0){
            ++m;
            q = m * (m + 1) - 2 * target;
        }
        return m;
    }
};
